File: 53-maximum-subarray/53-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        int max_sofar = -10001; 
        
        int max_tol = -10001; 
        
        
        for(const int& x : nums)
        {
            max_sofar = max(x, max_sofar + x); 
            
            max_tol = max(max_tol, max_sofar); 
        }
        
        return max_tol; 
    }
};

File: 53-maximum-subarray/53-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_so_far = float('inf') *-1
        max_sum = float('inf') *-1
        
        for x in nums:
            max_so_far = max(x, max_so_far + x)
            max_sum = max(max_sum, max_so_far)
            
        return max_sum
            
            
            
            